src/webgpu/util/math.ts

```diff
@@ -2030,22 +2030,26 @@ export function quantizeToF16(num: number): number {
   return quantizeToF16Data[0];
 }
 
-/** Statically allocate working data, so it doesn't need per-call creation */
-const quantizeToI32Data = new Int32Array(new ArrayBuffer(4));
-
 /** @returns the closest 32-bit signed integer value to the input */
 export function quantizeToI32(num: number): number {
-  quantizeToI32Data[0] = num;
-  return quantizeToI32Data[0];
+  if (num >= kValue.i32.positive.max) {
+    return kValue.i32.positive.max;
+  }
+  if (num <= kValue.i32.negative.min) {
+    return kValue.i32.negative.min;
+  }
+  return Math.round(num);
```

Comment:
Despite the function's comment, `quantizeToI32` wasn't rounding to closest, but was rounding towards zero:

```js
const F = 0.7;

const quantizeToI32Data = new Int32Array(new ArrayBuffer(4));
quantizeToI32Data[0] = F;
var message = `${quantizeToI32Data[0]}, ${Math.round(F)}`;

console.log(message)
```

This is a change of behaviour. Have you checked that all the places that call this expect the new behaviour?

Reply:
Changed these to use Math.trunc to replicate the behaviour. I have run all of the tests under webgpu:shader,execution,expression,* and the unittests, so I think this should be fine.